### PR TITLE
raised error with message for negative tof value

### PR DESCRIPTION
**Pull request by TheBuffer**
#1397 Added message to negative tof assert error

### CODE REVIEW

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Thanks for the pull request @TheBuffer!

However, this makes the change only in the `izzo` algorithm (while we also have `vallado`). And besides, the concept of "orbit epochs" doesn't exist at this low level function, it's only the `tof`, so I think the error message might be misleading.

What about moving the check upwards, to the `Maneuver.lambert` classmethod?

**Comment by astrojuanlu** [comment]
Also, this will need a test.

File: src/poliastro/maneuver.py [hunk 1]
```diff
@@ -185,6 +185,8 @@ def lambert(cls, orbit_i, orbit_f, method=lambert_izzo, short=True, **kwargs):
 
         # Time of flight is solved by subtracting both orbit epochs
         tof = orbit_f.epoch - orbit_i.epoch
+        if tof < 0:
+            raise ValueError("Time of Flight must be positive")
```
Comment (astrojuanlu) [inline src/poliastro/maneuver.py:189]: Can we have a more explanatory error message here? Mentioning the orbit epochs (like you initially did, but now we are on the right function)

File: src/poliastro/core/iod.py [hunk 2]
```diff
@@ -189,7 +189,6 @@ def izzo(k, r1, r2, tof, M, numiter, rtol):
     """
 
     # Check preconditions
-    assert tof > 0
```
Comment (astrojuanlu) [inline src/poliastro/core/iod.py:192]: Can we keep this assert?

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Just a couple of minor things and we're ready!

File: src/poliastro/maneuver.py (hunk 1 above)
Comment (TheBuffer) [reply src/poliastro/maneuver.py]: Sure!

**Review by astrojuanlu — APPROVED** [review]
Thanks a lot and congratulations for your first pull request! Merging

**Pull request merged**